apps/explorer/src/context.ts

```diff
@@ -37,14 +37,17 @@ export function useNetwork(): [string, (network: Network | string) => void] {
     }, [searchParams]);
 
     const setNetwork = (network: Network | string) => {
-        // When resetting the network, we reset the query client at the same time:
-        queryClient.cancelQueries();
-        queryClient.clear();
-
         setSearchParams({ network: network.toLowerCase() });
     };
 
-    useLayoutEffect(() => {
+    useEffect(() => {
+        // When the network changes (either from users changing the network manually or
+        // navigating back and forth between pages), we need to clear out our query cache
+        queryClient.cancelQueries();
```

Comment:
You should verify that this doesn't trigger double-loading on mount because it starts fetching, then gets invalidated here.